[#311] 1.0.1 Schema updates

Bjwebb opened the pull request:
This pull request batches several of the schema changes for 1.0.1 to make them easier to review together. A high level description of these changes can be found in [issue 311](https://github.com/open-contracting/standard/issues/311).

This pull request excludes changes to the versioned schema in 1.0.1, which are in a separate pull request at  https://github.com/open-contracting/standard/pull/304, as the diff is larger and easier to review separately.

<!---
@huboard:{"order":75.0,"milestone_order":305,"custom_state":"archived"}
-->

gabelula committed:
There is no statusDescription field in the standard

timgdavies committed:
[#300] BUG FIX: Updating descriptions and format validation of publisher

(In record-package-schema and release-package-schema).

Bjwebb committed:
[#247] 'releaseID' in description should be 'release ID'

Bjwebb committed:
[#295] Add null as an allowed value for contract/award status

Bjwebb committed:
[#295] Update the versionsed release validation schema

Bjwebb committed:
[#283] Remove ocdsVersion from Budget

This is not required, and there is no defined behaviour for having this
on an object like Budget.

Bjwebb committed:
[#283] Update the versionsed release validation schema

Bjwebb committed:
[#267] Notice codes in document type list should be camelCase

timgdavies commented:
We should drop https://github.com/open-contracting/standard/commit/6459ea31ae984edee3a3b479c0ec5b5c3ec4d3d3 out of this request.

That was included in error (my mistake for having that committed to a branch...)

Bjwebb committed:
Update schema uris to 1__0__1

Also uses a uri on the new standard.open-contracting.org, instead of the
old ocds.open-contracting.org.

Bjwebb commented:
Have rebased to remove that commit.

Bjwebb committed:
[#307] Merge @gabelula's codelist documentation fix
